workspaces/jenkins/plugins/scaffolder-backend-module-jenkins/src/action.ts

```diff
@@ -1,110 +0,0 @@
-/*
- * Copyright 2024 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
-import { InputError } from '@backstage/errors';
-import fetch from 'cross-fetch';
-import fs from 'fs/promises';
-import { resolveSafeChildPath } from '@backstage/backend-plugin-api';
-
-interface JenkinsCreateJobActionOptions {
-  config: {
-    getOptionalString: (key: string) => string | undefined;
-  };
-}
-
-const jenkinsCreateJobAction = (options: JenkinsCreateJobActionOptions) => {
```

Comment:
mmm you can't remove this action and replace with `createJob` as this will be a breaking change since the two actions are different. So keep this 😄

Reply:
I understand your point of view, but let me explain why i would to proceed with these changes:

- `jenkins:job:create-file` in this way we give more semantic to action and how its work, => it takes job name and a file, read it and then call jenkins
- the migration from old `jenkins:job:create` to new `jenkins:job:create-file` is very quick just see the readme (is possible to add more information about migration activity, in case isn't enough) and the user must update jenkins plugin to get the breacking change
- with jenkins client the action is more maintainable and readable for future improvement
- i introduce a new action `jenkins:job:create` to create a job just with 2 parameters `jobName` and `jobXml`

Anyway this is just my point of view, let me know if you are agree with me or you suggest another way to proceed 🙏

Reply:
> jenkins:job:create-file in this way we give more semantic to action and how its work, => it takes job name and a file, read it and then call jenkins

I'm not sure, it's a bit confusing having two actions doing almost the same job 😅
What about keeping only the existing `jenkins:job:create` adding `jobXML` as an optional input?

Reply:
Hi,
ok this could be a solution, but about the Jenkins integration could I maintain the client instead of existing logic?
In this case we haven't a  breaking change right?